index.js

```diff
@@ -361,6 +361,14 @@
 
   _Type['@@type'] = 'sanctuary-def/Type';
 
+  //  Type#fantasy-land/equals :: Type ~> Type -> Boolean
+  _Type.prototype['fantasy-land/equals'] = function(other) {
+    return Z.equals(this.type, other.type) &&
+           Z.equals(this.name, other.name) &&
+           Z.equals(this.url, other.url) &&
+           Z.equals(Z.toString(this), Z.toString(other));
```

Comment:
Do I understand correctly that we rely on the behaviour of `toString` for asserting equality of nested types?

Reply:
It's a simple approach but I now see that it ignores the `url` fields of nested types, leading to avoidable collisions between identically named types. I'll rework this.

Reply:
That. :)